### PR TITLE
Mostly fixing some GCC warnings

### PR DESCRIPTION
**Pull request by RyDroid**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,8 +20,8 @@ matrix:
 
 env:
   matrix:
-    - BUILD_TYPE="Debug"
-    - BUILD_TYPE="Release"
+    - BUILD_TYPE='Debug'
+    - BUILD_TYPE='Release'
```
Comment (auriamg) [inline .travis.yml]: Why that change? Seems completely unecessary

File: .travis.yml [hunk 2]
```diff
@@ -43,8 +49,8 @@ before_script:
   - free -mt
 
 script:
-  - mkdir "build"
-  - cd "build"
+  - mkdir build
+  - cd build
```
Comment (auriamg) [inline .travis.yml]: Why that change? Seems completely unecessary

File: lib/irrlicht/source/Irrlicht/CBlit.h [hunk 3]
```diff
@@ -513,8 +155,6 @@ static void executeBlit_TextureCopy_x_to_x( const SBlitJob * job )
 	}
 }
 
-/*!
-*/
```
Comment (auriamg) [inline lib/irrlicht/source/Irrlicht/CBlit.h]: Why that change? Seems completely unecessary

File: src/graphics/stk_scene_manager.cpp [hunk 4]
```diff
@@ -833,7 +833,7 @@ PROFILER_POP_CPU_MARKER();
                 ShadowCmdBuffer = (DrawElementsIndirectCommand*)glMapBufferRange(GL_DRAW_INDIRECT_BUFFER, 0, 10000 * sizeof(DrawElementsIndirectCommand), GL_MAP_WRITE_BIT | GL_MAP_UNSYNCHRONIZED_BIT | GL_MAP_INVALIDATE_BUFFER_BIT);
             }
 
-            for (unsigned i = 0; i < 4; i++)
+            for (unsigned char i = 0; i < 4; i++)
```
Comment (auriamg) [inline src/graphics/stk_scene_manager.cpp]: why??

Comment (RyDroid) [inline src/graphics/stk_scene_manager.cpp]: No wasting `sizeof(unsigned) - sizeof(char)`.

File: lib/irrlicht/source/Irrlicht/CBlit.h (hunk 3 above)
Comment (RyDroid) [inline lib/irrlicht/source/Irrlicht/CBlit.h]: Why keeping empty comment(s)? It is useless, and it takes 2 lines.

File: .travis.yml (hunk 2 above)
Comment (RyDroid) [inline .travis.yml]: Remove unecessary characters.

File: .travis.yml (hunk 1 above)
Comment (RyDroid) [inline .travis.yml]: No variable is searched with single quotes, that is not the case with double quotes.

File: src/graphics/stk_scene_manager.cpp (hunk 4 above)
Comment (hiker) [inline src/graphics/stk_scene_manager.cpp]: Doesn't make any difference, either it's in a register, or the loop is so slow (i.e. involving calls which force the register to be written back to memory) that it doesn't matter anyway.

File: lib/irrlicht/source/Irrlicht/CBlit.h (hunk 3 above)
Comment (hiker) [inline lib/irrlicht/source/Irrlicht/CBlit.h]: No one cares about two lines, it makes it potentially easier to merge with irrlicht updates later (less conflicts) - admittedly unlikely that we will ever do that though ;)

**Comment by hiker** [comment]
TBH, no change here will have any significant impact and just introduce noise to the commits, so closing this pull request.

Thank you!
   Joerg

**Pull request closed without merging**